main/src/com/google/refine/expr/functions/arrays/Zip.java

```diff
@@ -0,0 +1,115 @@
+met:
+
+    * Redistributions of source code must retain the above copyright
+notice, this list of conditions and the following disclaimer.
+    * Redistributions in binary form must reproduce the above
+copyright notice, this list of conditions and the following disclaimer
+in the documentation and/or other materials provided with the
+distribution.
+    * Neither the name of Google Inc. nor the names of its
+contributors may be used to endorse or promote products derived from
+this software without specific prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+"AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+*/
+
+package com.google.refine.expr.functions.arrays;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Properties;
+import java.util.Spliterator;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+
+import com.google.refine.expr.EvalError;
+import com.google.refine.grel.ControlFunctionRegistry;
+import com.google.refine.grel.EvalErrorMessage;
+import com.google.refine.grel.Function;
+import com.google.refine.grel.FunctionDescription;
+
+public class Zip implements Function {
+
+    @Override
+    public Object call(Properties bindings, Object[] args) {
+        if (args.length >= 2) {
+            List<Spliterator> iterators = new ArrayList<>();
+            for (Object v : args) {
+                if (v == null || !(v.getClass().isArray() || v instanceof List<?> || v instanceof ArrayNode)) {
+                    return new EvalError(
+                            EvalErrorMessage.expects_at_least_two_or_more_array_args(ControlFunctionRegistry.getFunctionName(this)));
+                } else if (v.getClass().isArray()) {
+                    iterators.add(Arrays.stream((Object[]) v).spliterator());
+                } else if (v instanceof ArrayNode) {
+                    ArrayNode a = (ArrayNode) v;
+                    List<Object> ziparg = new ArrayList<>();
+                    if (a.isArray()) {
+                        for (JsonNode objNode : a) {
+                            ziparg.add(objNode);
+                        }
+                    }
+                    iterators.add(ziparg.stream().spliterator());
```

Comment:
```suggestion
                    iterators.add(((ArrayNode) v).spliterator());
```

This can be simplified by using the `spliterator` method of ArrayNode.

Reply:
I added the loop as the iterator fails when processing Array node.

Reply:
It doesn't fail for me using your tests. Do you have a separate test which demonstrates the failure? Could you add it to the test suite please so that someone doesn't try to optimize this in the future. Thanks!